### PR TITLE
move postgres examples into a separate module

### PR DESCRIPTION
**Pull request by KevinJTaylor**

### CODE REVIEW

File: examples/postgres/Makefile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	gotestsum \
+		--format short-verbose \
+		--rerun-fails=5 \
+		--packages="./..." \
+		--junitfile ./build/test_unit.xml \
+		-- \
+		-coverprofile=./build/test_coverage.out \
+		-timeout=30m
+
+all: install lint test
```
Comment (KianYang-Lee) [inline examples/postgres/Makefile:38]: need to add newline here

File: examples/postgres/main.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package main
+
+func main() {
+
+}
```
Comment (KianYang-Lee) [inline examples/postgres/main.go:5]: If we are not allowing user to run it, we might as well just leave out this file

File: examples/postgres/build/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Will contain generated or build files created from makefile
```
Comment (KianYang-Lee) [inline examples/postgres/build/README.md:1]:
```suggestion
# Build

Will contain generated or build files created from `Makefile`
```

**Review by maknapik — DISMISSED** [review]
LGTM besides KY's comments

File: examples/postgres/flyway_postgres_test.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
```
Comment (mdelapenya) [inline examples/postgres/flyway_postgres_test.go:7]: I think this line should be moved next to the testcontainers imports

Comment (KevinJTaylor) [reply examples/postgres/flyway_postgres_test.go:7]: there is something odd about the way imports are organised ... I've seen projects which are alphabetical and ones which are more "golang" default. It must be a project setting as I don't manually do anything.

Comment (KevinJTaylor) [reply examples/postgres/flyway_postgres_test.go:7]: see if I organise imports I get get

image

Comment (KevinJTaylor) [reply examples/postgres/flyway_postgres_test.go:7]: I'm a bit confused by how this is setup tbh ... I played around with it by manually moving stuff, deleted the spaces, and then it miraculously started organising them in the way I would expect

**Review by KianYang-Lee — APPROVED** [review]

**Comment by KianYang-Lee** [comment]
@KevinJTaylor CI failing

File: examples/postgres/build/test_unit.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites tests="2" failures="0" errors="0" time="31.913450">
+	<testsuite tests="2" failures="0" time="21.259000" name="github.com/CyberOwlTeam/flyway/examples/postgres" timestamp="2024-07-01T10:21:53+01:00">
+		<properties>
+			<property name="go.version" value="go1.22.3 darwin/amd64"></property>
+		</properties>
+		<testcase classname="github.com/CyberOwlTeam/flyway/examples/postgres" name="TestFlyway_postgres" time="16.180000"></testcase>
+		<testcase classname="github.com/CyberOwlTeam/flyway/examples/postgres" name="ExampleFlyway_postgres" time="4.620000"></testcase>
+	</testsuite>
+</testsuites>
```
Comment (KianYang-Lee) [inline examples/postgres/build/test_unit.xml:10]: add newline

File: examples/postgres/example_flyway_postgres_test.go [hunk 6]
```diff
@@ -13,7 +13,7 @@ import (
 	tcnetwork "github.com/testcontainers/testcontainers-go/network"
 )
 
-func ExampleRunContainer() {
+func ExampleFlyway_postgres() {
```
Comment (KianYang-Lee) [inline examples/postgres/example_flyway_postgres_test.go:16]: maybe uppercase for `postgres` ? looks more consistent. Personal preference though

**Review by KianYang-Lee — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# flyway
+# Flyway Test Container
```
Comment (mdelapenya) [inline README.md:1]:
```suggestion
# Flyway module for Testcontainers for Go
```

**Review by mdelapenya — APPROVED** [review]
I like having the postgres examples in a separate module, as it will reduce the dependencies surface for the core module.

LGTM, thanks!

File: examples/postgres/build/test_unit.xml (hunk 5 above)
Comment (mdelapenya) [reply examples/postgres/build/test_unit.xml:10]: Do we need to add the xUnit files to the repo? I think we can ignore them at the gitignore file

File: examples/postgres/example_flyway_postgres_test.go (hunk 6 above)
Comment (KianYang-Lee) [reply examples/postgres/example_flyway_postgres_test.go:16]: Thanks good to know @mdelapenya

**Pull request merged**